internal/util/diagtree/diagtree.go

```diff
@@ -180,20 +180,23 @@ func (m *Node) severity() string {
 // The severity of a node.
 //
 // Nodes with their own (non-None) severity are always displayed on their own level.
-type Severity string
+type Severity struct{ s string }
 
-const (
-	None   Severity = ""
-	Info   Severity = "`🟢`"
-	Warn   Severity = "`🟡`"
-	Danger Severity = "`🔴`"
+var (
+	None   = Severity{""}
+	Info   = Severity{"`🟢`"}
+	Warn   = Severity{"`🟡`"}
+	Danger = Severity{"`🔴`"}
 )
 
+func (s Severity) String() string {
```

Comment:
See comment above; if we want to keep this getter function, I think calling it "GetLevel" or "GetSeverity" would be more descriptive.

Reply:
`String` isn't used as a getter function here, it's an override on how a struct should be displayed to the user. (see [`fmt.Stringer`](https://pkg.go.dev/fmt#Stringer)).

Note that `String` is never called when we want to check the level. For that we can compare directly:

https://github.com/pulumi/schema-tools/blob/21dae2711935ef927ab66c7cbb745cdf9092160d/internal/util/diagtree/diagtree.go#L170-L170

Reply:
Ah, gotcha! Thanks for explaining. Makes sense to use String() then.

As for enums... I'm pretty sure I've used enums in Go before; something like:

```
type Severity int

const (
    None iota
    Info
    Warn
    Danger
)
```

I may be missing your concern about type safety though?

Reply:
We had that before:
https://github.com/pulumi/schema-tools/blob/c7ca835cdec9f91bf8b6ac80ebee16bb5607a5ed/internal/util/diagtree/diagtree.go#L185-L190

The problem was that the literal `"missing type"` was a valid instance of `Severity` (which caused the previous bug).

If we go with your suggestion, then the numeric literal `-7` is a valid instance of `Severity`. This is less likely to cause problems, but still allows a user to construct an invalid enum instance.

AFAIK, it is impossible to construct a severity level outside of the `diagtree` package that is not one of `None`, `Info`, `Warn` or `Danger`. This is the property I am trying to preserve.